zarr/tests/test_storage.py

```diff
@@ -1248,10 +1248,6 @@ def test_read_only(self):
 
         assert store[self.root + 'foo'] == b"bar"
 
-        filepath = os.path.join(path, self.root + "foo")
-        with pytest.raises(ValueError):
-            self.create_store(path=filepath, mode='r')
```

Comment:
...and its corresponding test.

This test did not do anything other than test the error above. `self.create_store` does not actually attempt to write any data or list the store.

With the check gone, users will now see an error from fsspec if they try to do an operation that requires listing on an unlistable FSSpec store, as they would already on a generic MutableMapping-based store.